zerver/views/typing.py

```diff
@@ -12,22 +17,21 @@
 from zerver.models import UserProfile
 
 VALID_OPERATOR_TYPES = ["start", "stop"]
-VALID_RECIPIENT_TYPES = ["direct", "stream"]
+VALID_RECIPIENT_TYPES = ["direct", "stream", "message_edit", "direct_message_edit"]
 
 
 @has_request_variables
 def send_notification_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    req_type: str = REQ(
-        "type", str_validator=check_string_in(VALID_RECIPIENT_TYPES), default="direct"
-    ),
+    req_type: str = REQ("type", str_validator=check_string_in(VALID_RECIPIENT_TYPES)),
```

Comment:
What's the thinking around removing the `default` property? I'm pretty sure that's required for existing mobile/terminal app versions that predate stream typing notifications.

Reply:
Previously, when typing notifications were available only to stream message edits and direct message edits, when the POST doesn't include `type="stream"`, it is taken as `type="direct"` by default. 

Now that we are adding extra types for message edit in these POST, I felt it would be better if the POST of direct message edits rather explicitly contained the `type="direct"` property, rather than `default="direct"`

Reply:
So the intent is that any modern client should always pass this parameter. But clients that predate stream typing notifications and thus don't pass this parameter exist, including actually the current version of the mobile apps, and they need the default of "direct" to function.

You can add a comment explaining this to help others not "simplify" by removing it -- but it's also hinted in the API documentation for the endpoint, which is always worth reading when changing the API.